### PR TITLE
Update the `Readme.md` with build and debug instructions

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Update the `Readme.md` with build and debug instructions.

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/15521

### How to test this PR?

N/A

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by ibuziuk — COMMENTED** [review]
@vinokurig could you please update the `What is Che server` part as part of this PR? would be great to review each section of README.md and update / remove the parts which are not relevant anymore

File: README.md [hunk 1]
```diff
@@ -1,27 +1,44 @@
 # What is Che server
-Che Server is a core component of the [Eclipse Che](https://github.com/eclipse/che/). This component is responsible for creation and managing of Che workspaces, but will some day be replaced by the [Dev Workspace Operator](https://github.com/devfile/devworkspace-operator). 
+Che Server provides an Api to retrieve devfile content from the given repositories such as GitHub GitLab Bitbucket and Azure.
```
[the remark this answers is not in the document]
Comment (vinokurig) [reply README.md:2]: done

File: README.md [hunk 2]
```diff
@@ -1,27 +1,48 @@
...
```
Comment (max-cx) [inline README.md:2]: If I'm correct, API is an abbreviation, so it needs to be capitalized, plus a few more suggested changes to this sentence:
```suggestion
Che Server provides an API to retrieve devfile content from repositories hosted on GitHub, GitLab, Bitbucket, and Microsoft Azure Repos.
```

Comment (ibuziuk) [reply README.md:2]: nice suggestion :+1:

File: README.md [hunk 3]
```diff
@@ -1,27 +1,48 @@
...
+Apart from that Che Server provides an Api to provision Kubernetes namespaces.
```
Comment (ibuziuk) [inline README.md:3]: @vinokurig let's add it to the first sentence e.g. `Che Server provides an API for managing Kubernetes namespaces and retrieving  devfile content from repositories hosted on GitHub, GitLab, Bitbucket, and Microsoft Azure Repos`

Comment (vinokurig) [reply README.md]: done

File: README.md [hunk 4]
```diff
@@ -1,27 +1,48 @@
...
+### Factory flow implementations
+- wsmaster/che-core-api-factory-azure-devops
+- wsmaster/che-core-api-factory-bitbucket
+- wsmaster/che-core-api-factory-bitbucket-server
+- wsmaster/che-core-api-factory-github
+- wsmaster/che-core-api-factory-gitlab
+- wsmaster/che-core-api-factory-shared
+### Kubernetes namespace provisioning
+- infrastructures/kubernetes
+- infrastructures/infrastructure-factory
```
[the remark this answers is not in the document]
Comment (vinokurig) [reply README.md]: fixed

File: README.md [hunk 5]
```diff
@@ -1,27 +1,49 @@
...
-- ['dockerfiles'](https://github.com/eclipse-che/che-server/tree/main/dockerfiles) - directory contains image Dockerfile for Che Server, as well as additional images.
-- ['core'](https://github.com/eclipse-che/che-server/tree/main/core) - core and utility modules for Che.
-- ['wsmaster'](https://github.com/eclipse-che/che-server/tree/main/wsmaster) - primary modules of the Che Server API.
-- ['multiuser'](https://github.com/eclipse-che/che-server/tree/main/multiuser) - modules related to multiuser implementation of Che.
-- ['infrastructure'](https://github.com/eclipse-che/che-server/tree/main/infrastructure) - implementations for the underlying infrastructure, on which Che is running (Kubernetes, Openshift, etc.)
-- ['deploy'](https://github.com/eclipse-che/che-server/tree/main/deploy) - deployment files for Helm installation.
-- ['typescript-dto'](https://github.com/eclipse-che/che-server/tree/main/typescript-dto) module, that provides DTO objects for typescript projects that may depend on Che Server, such as Che Theia.
+Che Server is mostly a Java web application deployed on an Apache Tomcat server in a container. The relevant modules are:
```
Comment (max-cx) [inline README.md:6]: @vinokurig, what do you mean by `relevant` here? These modules are relevant in which context, to whom, why?

File: README.md [hunk 6]
```diff
@@ -1,27 +1,49 @@
...
-- ['core'](https://github.com/eclipse-che/che-server/tree/main/core) - core and utility modules for Che.
-- ['wsmaster'](https://github.com/eclipse-che/che-server/tree/main/wsmaster) - primary modules of the Che Server API.
-- ['multiuser'](https://github.com/eclipse-che/che-server/tree/main/multiuser) - modules related to multiuser implementation of Che.
-- ['infrastructure'](https://github.com/eclipse-che/che-server/tree/main/infrastructure) - implementations for the underlying infrastructure, on which Che is running (Kubernetes, Openshift, etc.)
-- ['deploy'](https://github.com/eclipse-che/che-server/tree/main/deploy) - deployment files for Helm installation.
-- ['typescript-dto'](https://github.com/eclipse-che/che-server/tree/main/typescript-dto) module, that provides DTO objects for typescript projects that may depend on Che Server, such as Che Theia.
+Che Server is mostly a Java web application deployed on an Apache Tomcat server in a container. The relevant modules are:     
+### oAuth1 / oAuth2 API implementations
```
Comment (max-cx) [inline README.md:7]:
```suggestion
### OAuth1 / OAuth2 API implementations
```

File: README.md [hunk 7]
```diff
@@ -1,27 +1,49 @@
...
+- wsmaster/che-core-api-factory-shared
+### Kubernetes namespace provisioning
+- infrastructures/kubernetes
+- infrastructure/openshift
+- infrastructures/infrastructure-factory
+
+Other modules are deprecated and will be removed in the future.
 
 # Build requirements
 - Apache Maven 3.6.3 or Higher
```
Comment (max-cx) [inline README.md:28]: Our style guide suggests using the word `earlier` and `later`/`latest` when describing software versions:
```suggestion
- Apache Maven 3.6.3 or later
```

File: README.md [hunk 8]
```diff
@@ -1,27 +1,49 @@
...
 - JDK Version 11
 - Podman or Docker (required for running integration tests)
 
-# Build and debug
-Run `mvn clean install` to build 
-Activate a faster profile build by adding `-Pfast`
-To debug, run `mvn clean install -X` and connect your IDE to the debug port
+# Sources build
+Run `mvn clean install` to build. Activate a faster profile build by adding `-Pfast`
```
Comment (max-cx) [inline README.md:33]:
```suggestion
Run `mvn clean install` to build. Activate a faster profile build by adding `-Pfast`.
```

File: README.md [hunk 9]
```diff
@@ -1,27 +1,49 @@
...
+
+# Image build and push
+1. Go to the `dockerfiles` directory.
+2. Run `./build.sh`.
+3. Tag the **che-server** image with your account `docker tag quay.io/eclipse/che-server:next <docker registry>/<your account>/che-server:next`.
```
Comment (max-cx) [inline README.md:38]:
```suggestion
3. Tag the **che-server** image with your account: `docker tag quay.io/eclipse/che-server:next <docker registry>/<your account>/che-server:next`.
```

File: README.md [hunk 10]
```diff
@@ -1,27 +1,49 @@
...
+4. Push the **che-server** image to your account `docker push <docker registry>/<your account>/che-server:next`.
```
Comment (max-cx) [inline README.md:39]:
```suggestion
4. Push the **che-server** image to your account: `docker push <docker registry>/<your account>/che-server:next`.
```

File: README.md [hunk 11]
```diff
@@ -1,27 +1,49 @@
...
+
+# Debug
+1. Deploy Che to [an openshift cluster](https://www.eclipse.org/che/docs/stable/administration-guide/installing-che-on-openshift-using-cli/) or [minikube](https://www.eclipse.org/che/docs/stable/administration-guide/installing-che-on-minikube/). Use the previously built image: `chectl server:start --platform=<openshift / minikube> --cheimage=<docker registry>/<your account>/che-server:next`
```
Comment (max-cx) [inline README.md:42]:
```suggestion
1. Deploy Che to [a Red Hat OpenShift](https://www.eclipse.org/che/docs/stable/administration-guide/installing-che-on-openshift-using-cli/) or [Minikube](https://www.eclipse.org/che/docs/stable/administration-guide/installing-che-on-minikube/) cluster by using a previously built image: `chectl server:start --platform=<openshift / minikube> --cheimage=<docker registry>/<your account>/che-server:next`  
```

File: README.md [hunk 12]
```diff
@@ -1,27 +1,49 @@
...
+
+# Image build and push
+1. Go to the `dockerfiles` directory.
+2. Run `./build.sh`.
+3. Tag the **che-server** image with your account `docker tag quay.io/eclipse/che-server:next <docker registry>/<your account>/che-server:next`.
+4. Push the **che-server** image to your account `docker push <docker registry>/<your account>/che-server:next`.
+
+# Debug
+1. Deploy Che to [an openshift cluster](https://www.eclipse.org/che/docs/stable/administration-guide/installing-che-on-openshift-using-cli/) or [minikube](https://www.eclipse.org/che/docs/stable/administration-guide/installing-che-on-minikube/). Use the previously built image: `chectl server:start --platform=<openshift / minikube> --cheimage=<docker registry>/<your account>/che-server:next`  
+2. Enable local debug of Eclipse Che server: `chectl server:debug`.
```
Comment (max-cx) [inline README.md:43]:
```suggestion
2. Enable local debug of the Eclipse Che server: `chectl server:debug`.
```

File: README.md [hunk 13]
```diff
@@ -1,27 +1,49 @@
...
+3. In your IDE create a new Remote JVM Debug configuration on `localhost:8000`.
```
Comment (max-cx) [inline README.md:44]:
```suggestion
3. In your IDE, create a new Remote JVM Debug configuration on `localhost:8000`.
```

File: README.md [hunk 14]
```diff
@@ -52,7 +74,7 @@ Downstream builds can be found at the link below, which is _internal to Red Hat_
 # Join the community
 
 The Eclipse Che community is globally reachable through public chat rooms, mailing list and weekly calls.
-See https://www.eclipse.org/che/docs/che-7/overview/introduction-to-eclipse-che/#_joining_the_community
+See https://www.eclipse.org/che/docs/stable/overview/introduction-to-eclipse-che/#_joining_the_community
```
Comment (max-cx) [inline README.md:77]:
```suggestion
See https://www.eclipse.org/che/docs/stable/overview/introduction-to-eclipse-che/#_joining_the_community.
```

**Review by ibuziuk — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (vinokurig) [reply README.md:6]: It means that since we have the devworkspace operator, we use only those modules, others are deprecated.

**Review by max-cx — APPROVED** [review]

**Pull request merged**